### PR TITLE
Allow to use custom upstream nameservers.

### PR DESCRIPTION
**Pull request by ypid**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -168,6 +168,14 @@ dnsmasq_search: [ '{{ dnsmasq_domain }}', '{{ ansible_domain }}' ]
 dnsmasq_cname: {}
 
 
+# .. envvar:: dnsmasq_nameservers
+#
+# List of upstream DNS resolvers where dnsmasq should forward it’s DNS queries
+# which it can not answer by itself to. If set to ``False`` it will use the
+# systems nameservers.
+dnsmasq_nameservers: False
```
Comment (drybjed) [inline defaults/main.yml]: If it's a list, it should be defined as `[]` to avoid issues with variable type conversion.

Comment (ypid) [inline defaults/main.yml]: Makes sense.

**Comment by tomislacker** [comment]
@drybjed have you ever seen where dnsmasq doesn't reply in a timely manner, when using multiple upstream servers, when at least one of the upstreams replies with `SERVFAIL`? I'm currently fighting with that and considering making a PR to add the `strict-order` parameter *somewhere* since it resolved my particular failure condition.

**Comment by drybjed** [comment]
@tomislacker No, I haven't noticed that behaviour. From the documentation it seems that `strict-order` is very situational, the default behaviour of asking DNS servers randomly taking note of which ones are up sounds better as a default. You should still be able to include it in your configuration by adding this option to one of `dnsmasq__options` YAML text block variables.

**Pull request merged**